src/ui/application-window.vala

```diff
@@ -251,6 +254,23 @@ namespace Peek.Ui {
       record_button.set_label (_("Record as %s").printf (format_name));
       pop_format.hide ();
     }
+    
+    private void update_time () {
+      int seconds=0;
+      Timeout.add_seconds_full (GLib.Priority.LOW,1, () => {
```

Comment:
So let's see how this works out. If we find out the time will vary too much for longer recordings, we can first maybe try to increase the priority here or do the time calculation on each call based on a start time.